### PR TITLE
Adds an HUD button to change your z-level

### PR DESCRIPTION
**Pull request by TheSmallBlue**
## About The Pull Request

Adds a quick shortcut to the "Move up" and "Move down" verbs via a button on the HUD!

Gnarly as hell features include:
- Implemented on humans, ghosts, cyborgs, AIs and Xenomorphs!
- One button, two actions! Left click goes up, right click goes down.
- Each UI style has its own themed sprite! (I made them myself and I'm not good spriting but I like how they ended up)
- Has a tooltip! (Wow!)

Demo video (has old button placement, look below to see the new placement):

https://youtu.be/Mlw6C_ta50g

<details>
<summary>Mobs screenshots</summary>

AI:
image

Ghost:
image

Larva/Xenomorph:
image

Cyborg:
image

</details>

<details>
<summary>UI styles screenshots (old placement)</summary>

Midnight:
image

Clockwork:
image

Detective:
image

Glass:
image

Operative:
image

Plasmafire:
image

Retro:
image

Slimecore:
image

Trasen-knox:
image

</details>

**New placement:**

<details>

image

</details>

## Why It's Good For The Game

We already have a bunch of maps with multiple Z-levels (Northstar, Tramstation, Icebox) and I'm sure there will be more to come.
As these maps become more common, the ability to move up/down z-levels is bound to become important, if not essential. So I thought that, instead of hiding this ability behind a tab or a hotkeys menu, why not have it front and center?

Also, I remember a long time ago there was a push to make the verbs menu on the top right obsolete, this helps to get us to that goal if we're still doing it.

(This whole thing was inspired by me seeing what I assume to be a not-so-new but not-so-experienced player being confused as to how to go down those exact stairs on Tramstation's science when the gravgen was broken, and my inability to be able to explain it to them, partly because i couldnt remember what the key was called, and partly because i was a mime.)
## Changelog
:cl:
qol: added an HUD button to go up and down floors
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
...Does that button placement work with non-widescreen?

**Comment by TheSmallBlue** [comment]
Fuck

image

**Comment by TheSmallBlue** [comment]
New position!
<details>

Widescreen:
image

Not widescreen:
image

</details>

It still looks fine in every other mob

**Comment by Helg2** [comment]
Maybe put it a bit closer to the Resist button so it looks better?

**Comment by Elli-Skala** [comment]
Good PR, changing z level and looking up and down always felt weird hidden away in the tabs for something that was a physical action

**Comment by LemonInTheDark** [comment]
This isn't really your fault but I'm not really sure how to feel about this. The verb sucks, but I also really want to avoid cluttering the UI.

**Comment by TheSmallBlue** [comment]
> This isn't really your fault but I'm not really sure how to feel about this. The verb sucks, but I also really want to avoid cluttering the UI.

I specifically made it the smallest, most least intrusive button I could get away with for this reason. It is true the UI is already cluttered as it is, but I do seriously think moving up and down is essential enough that it's worth sacrificing 32~ pixels of hud space.

**Comment by Ghommie** [comment]
> This isn't really your fault but I'm not really sure how to feel about this. The verb sucks, but I also really want to avoid cluttering the UI.

We should look into collapsible UI buttons. Perhaps not the up/down button, but all the ones beside it could be squashed into a single one that opens up a radial menu. Also there's more vertical space to use than horizontal, especially with widesrcreen enabled. Also a trasparency slider pref would be nice. I often feel like the glass ui style isn't trasparent enough for my tastes.

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -127,6 +127,33 @@
...
+
+/atom/movable/screen/floor_menu/Initialize(mapload)
+	. = ..()
+	register_context()
+
+/atom/movable/screen/floor_menu/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+
+	context[SCREENTIP_CONTEXT_LMB] = "Go up a floor"
+	context[SCREENTIP_CONTEXT_RMB] = "Go down a floor"
```
Comment (Ghommie) [inline code/_onclick/hud/screen_objects.dm:144]:
```suggestion
	if(isobserver(user))
		context[SCREENTIP_CONTEXT_ALT_LMB] = "Go down a floor"
	else
		context[SCREENTIP_CONTEXT_RMB] = "Go down a floor"
```

**Comment by Hemlock32** [comment]
Kinda, only works in one direction as a ghost, which is the main time one would even be using this. There's a reason Eris has two buttons instead of one

**Comment by Ghommie** [comment]
Yeeea, It should pop up the up-down radial for ghosts.

**Comment by TheSmallBlue** [comment]
> Kinda, only works in one direction as a ghost, which is the main time one would even be using this. There's a reason Eris has two buttons instead of one

Alt-click the button, it will also make you to down. I'll see about adding a better way later on

**Pull request merged**